website_product_subscription/controllers/subscribe_form.py

```diff
@@ -86,6 +92,45 @@ def _validate_email_unique(self, email):
                 "address. Please login before fill in the form"
             )
 
+    def _validate_subscriber_email_unique(self, email):
+        other_user = (
+            request.env["res.users"]
+            .sudo()
+            .search([("login", "=", email)], limit=1)
+        )
+        subscriber_address_checked = self.qcontext.get(
+            "subscriber_address_checked", False
+        )
+        subscriber_address_checked_for = self.qcontext.get(
+            "subscriber_address_checked_for", ""
+        )
+
+        if (
+            subscriber_address_checked_for != email
+            or not subscriber_address_checked
+        ):
+            if other_user:
+                # check parameters
+                self.qcontext["subscriber_address_checked"] = True
+                self.qcontext["subscriber_address_checked_for"] = email
+
+                # load subscriber address
+                partner = other_user.partner_id
+                self.qcontext["subscriber_firstname"] = partner.firstname
+                self.qcontext["subscriber_lastname"] = partner.lastname
+                self.qcontext["subscriber_street"] = partner.street
+                self.qcontext["subscriber_zip"] = partner.zip
+                self.qcontext["subscriber_city"] = partner.city
+                self.qcontext["subscriber_country_id"] = partner.country_id.id
+
+                # message
+                self.qcontext["message_to_user"] = _(
+                    "There is an existing account for the person to which you "
+                    "are addressing this subscription. Hereunder is the "
+                    "address we have for this person. Please check the "
+                    "address and confirm again. "
+                )
```

Comment:
Is it really GDPR to show the address of a person ?

Reply:
oui pas sûr qu'afficher l'adresse pour une personne soit respectueux de la loi sur la vie privée.

Reply:
mmhf good point. @vdewulf  What do you think? Should i put back the behavior I first programmed, that is, ask to confirm the input address rather than the address in the database?

Reply:
Virginie voit avec Laurence.